### PR TITLE
Fixed projection call

### PR DESCRIPTION
**Pull request by rechkalov**
This change is [Reviewable](https://reviewable.io/reviews/2gis/nuclear-aggregates-layer/24)

### CODE REVIEW

**Comment by maximpashuk** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/2gis/nuclear-aggregates-layer/24)*

**Comment by denisivan0v** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[Storage.LinqToDB.Tests/LinqToDBDomainContextSpecs.cs, line 84 at r1](https://reviewable.io:443/reviews/2gis/nuclear-aggregates-layer/24#-KRcXGjRqFl0KTM9GJ_8:-KRcXGjRqFl0KTM9GJ_9:b-8pz4f4) ([raw file](https://github.com/2gis/nuclear-aggregates-layer/blob/62b5971d086a5e1ef72460df1c4f80a71d356d51/Storage.LinqToDB.Tests/LinqToDBDomainContextSpecs.cs#L84)):*
> ```Smalltalk
> 
>             It should_not_throw_exceptions = () => _exception.Should().Be(null);
>             It should_return_entity = () => _entityId.Should().NotBe(null);
> ```

стоит сравнить с добавленной Entity

---

*[Storage.LinqToDB/Connections/LinqToDBQueryProviderDecorator.cs, line 58 at r1](https://reviewable.io:443/reviews/2gis/nuclear-aggregates-layer/24#-KRcXMZQDp4GmvVw2NI1:-KRcXMZRgLiCUA9hFKKd:b-e7r6c7) ([raw file](https://github.com/2gis/nuclear-aggregates-layer/blob/62b5971d086a5e1ef72460df1c4f80a71d356d51/Storage.LinqToDB/Connections/LinqToDBQueryProviderDecorator.cs#L58)):*
> ChildProviderForQueryable
А что этим достигается?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/2gis/nuclear-aggregates-layer/24)*

**Comment by rechkalov** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[Storage.LinqToDB.Tests/LinqToDBDomainContextSpecs.cs, line 84 at r1](https://reviewable.io:443/reviews/2gis/nuclear-aggregates-layer/24#-KRcXGjRqFl0KTM9GJ_8:-KRcXyIq5lV7b3oUpho-:b-4acit8) ([raw file](https://github.com/2gis/nuclear-aggregates-layer/blob/62b5971d086a5e1ef72460df1c4f80a71d356d51/Storage.LinqToDB.Tests/LinqToDBDomainContextSpecs.cs#L84)):*
<details><summary><i>Previously, denisivan0v (Denis Ivanov) wrote…</i></summary>
> стоит сравнить с добавленной Entity
</details>
На самом деле не принципиально - важно только, что не null. Но ладно, исправлю.

---

*[Storage.LinqToDB/Connections/LinqToDBQueryProviderDecorator.cs, line 58 at r1](https://reviewable.io:443/reviews/2gis/nuclear-aggregates-layer/24#-KRcXMZQDp4GmvVw2NI1:-KRcYAmrI7292Og5Gr-2:be68f8r) ([raw file](https://github.com/2gis/nuclear-aggregates-layer/blob/62b5971d086a5e1ef72460df1c4f80a71d356d51/Storage.LinqToDB/Connections/LinqToDBQueryProviderDecorator.cs#L58)):*
<details><summary><i>Previously, denisivan0v (Denis Ivanov) wrote…</i></summary>
> > ChildProviderForQueryable
> А что этим достигается?
</details>
Достигается то, что мы больше не пытаемся извлечь long используя IQueryProvider, который linq2db создал для работы с Entity. Ну или как-то так. В меру моего понимания.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/2gis/nuclear-aggregates-layer/24)*

**Comment by rechkalov** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[Storage.LinqToDB/Connections/LinqToDBQueryProviderDecorator.cs, line 58 at r1](https://reviewable.io:443/reviews/2gis/nuclear-aggregates-layer/24#-KRcXMZQDp4GmvVw2NI1:-KRcZ20z7iU6JVUjBIf2:box9veq) ([raw file](https://github.com/2gis/nuclear-aggregates-layer/blob/62b5971d086a5e1ef72460df1c4f80a71d356d51/Storage.LinqToDB/Connections/LinqToDBQueryProviderDecorator.cs#L58)):*
<details><summary><i>Previously, rechkalov (a.rechkalov) wrote…</i></summary>
> Достигается то, что мы больше не пытаемся извлечь long используя IQueryProvider, который linq2db создал для работы с Entity. Ну или как-то так. В меру моего понимания.
</details>
Вообще, поразительно, что с этим столкнулись только сегодня.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/2gis/nuclear-aggregates-layer/24)*

**Comment by denisivan0v** [comment]
Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/2gis/nuclear-aggregates-layer/24)*

File: Storage.LinqToDB.Tests/LinqToDBDomainContextSpecs.cs [hunk 1]
```diff
@@ -56,6 +56,34 @@ class When_call_nongeneric_GetQueryableSource
...
+                                                                   new TransactionOptions
+                                                                   {
+                                                                       IsolationLevel = IsolationLevel.ReadCommitted,
+                                                                       Timeout = TimeSpan.Zero
+                                                                   },
+                                                                   new NullPendingChangesHandlingStrategy());
+            };
+
+            private Because of = () => _exception = Catch.Exception(() =>
+                _entityId = _readableDomainContext.GetQueryableSource<Entity>().Select(x => x.Id).Single());
```
Comment (denisivan0v) [inline Storage.LinqToDB.Tests/LinqToDBDomainContextSpecs.cs]: Это раньше не работало

**Pull request merged**